api/facadeversions.go

```diff
@@ -18,7 +18,7 @@ var facadeVersions = map[string]int{
 	"AllModelWatcher":              4,
 	"AllWatcher":                   3,
 	"Annotations":                  2,
-	"Application":                  16,
+	"Application":                  17,
```

Comment:
Thinking about this line change, I feel like now that we have actually gone to 3.0 and have dropped support for old versions, all of these single integers actually need to become tuples with start/end, or lists with explicit numbers.

I'm happiest with tuples, and also very happy to not introduce this in 3.0, but make sure we start getting it right moving forward. (We sort of cheated heavily by just rejecting any target that is <3, but we *should* support a range of facade versions)

Reply:
Agreed.